### DISCUSSION TITLE
API to list enabled external auth providers?

### DISCUSSION THREAD

**Discussion by jhollowayj** [pocketbase/pocketbase#5518] [Q&A] [answered]
First off, I absolutely love this project and am so happy to have found it.  You easily saved me 4 weeks of having to build all of this myself.  Thank you for making so much available!

-----

I have a use case where the list of oauth providers will be different per deployment, but I want the the frontend code I bundle with it to stay the same.
To enable this with respect to the login page, I'd like the frontend to query for the current list of available providers and add `Login with {{DisplayName}}` buttons for each one.

I didn't see any public (non-authenticated) APIs that show this list of enabled external authentication providers.  Perhaps I missed it.  `/settings` requires to be authenticated previously.  I don't think I saw another place calling `NamedAuthProviderConfigs()`, but maybe I was searching for the wrong thing.

Here's my work around I came up with for now:

```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	e.Router.GET("/api/external-auth-provider-list", func(c echo.Context) error {
		// Custom return type, with the limited data I cared about
		type ExternalProvider struct {
			Id          string `json:"id"`
			DisplayName string `json:"displayName"`
		}

		ret := []ExternalProvider{}

		// Filter the list pf providers down to only the enabled ones
		for id, cfg := range app.Settings().NamedAuthProviderConfigs() {
			if cfg.Enabled {
				ret = append(ret, ExternalProvider{id, cfg.DisplayName})
			}
		}

		return c.JSON(http.StatusOK, ret)
	})

	return nil
})
```

Is there a better way of doing this?

If not, I'm happy to put a PR up to add this sort of functionality if there's interest in supporting this functionality out of the box.

**Comment by ganigeorgiev** [comment, marked as answer]
You can call the ["List auth methods" Web API](https://pocketbase.io/docs/api-records/#list-auth-methods).

**Reply by jhollowayj** [reply]
Awesome!  I feel like an idiot for not being able to find that on my own.  :D
